File: common/src/main/kotlin/io/edanni/rndl/common/domain/entity/Trip.kt
package io.edanni.rndl.common.domain.entity

import org.threeten.bp.*
import java.math.BigDecimal

data class Trip(
        // Database
        override val id: Long? = null,
        override val createdAt: OffsetDateTime? = null,
        override val updatedAt: OffsetDateTime? = null,
        val startTimestamp: Long? = null,
        val duration: LocalTime? = null,
        val distance: Double? = null,
        val averageSpeed: Int? = null,
        val maximumSpeed: Int? = null,
        val economy: Double? = null,
        val fuelUsed: BigDecimal? = null,
        val vehicleId: Long? = null,
        // Entity
        val vehicle: Vehicle? = null,
        val entries: List<Entry> = emptyList()
) : Entity {
    val startTime: LocalDateTime? = if (startTimestamp != null) {
        LocalDateTime.ofInstant(Instant.ofEpochMilli(startTimestamp), ZoneId.systemDefault())
    } else {
        null
    }
}

File: webapp/src/api/entities/vehicle/Trip.ts
import Entity from '../Entity';
import Vehicle from './Vehicle';
import Entry from './Entry';

export default interface Trip extends Entity {
    startTimestamp: number;
    duration: string;
    distance: number;
    averageSpeed: number;
    maximumSpeed: number;
    economy: number;
    fuelUsed: number;
    vehicleId: number;

    vehicle: Vehicle;
    entries: Entry[];
    startTime: string;
}